mev/poll.go

```diff
@@ -0,0 +1,62 @@
+package mev
+
+import (
+	"context"
+	types "github.com/m4ksio/silo-mev-protobuf-go/mev/v1"
+	"github.com/tendermint/tendermint/libs/log"
+	"google.golang.org/grpc"
+	"time"
+)
+
+type Poller struct {
+	client            types.BundleProviderClient
+	keeper            *Keeper
+	lastBlockProvider func() int64
+	logger            log.Logger
+}
+
+func (p *Poller) run() {
+
+	lastHeight := p.lastBlockProvider()
+
+	bundles, err := p.client.GetBundles(context.Background(), &types.GetBundlesRequest{MinBlockHeight: uint64(lastHeight)})
+	if err != nil {
+		p.logger.Error("Error while querying SILO server for bundles", "err", err)
+		return
+	}
+	// TODO validate data, don't trust height
+	for height, bundles := range bundles.Bundles {
+		p.keeper.AddBundles(int64(height), bundles.Bundles)
+	}
+
+}
+
+func NewPoller(logger log.Logger, config Config, keeper *Keeper, lastBlockProvider func() int64) (*Poller, error) {
+
+	logger.Info("Starting bundle provider poller")
+
+	// TODO secure grpc connection
+	grpcConn, err := grpc.DialContext(context.Background(), config.ServerAddr, grpc.WithInsecure())
```

Comment:
can we pass a real context here so things can exit reasonably

Reply:
I agree that some shutdown mechanism should be added, but I cannot find one actually used by the application. `sdk.Context` contains `context.Context` which can be used, but I don't think this ever gets cancelled anywhere. Similarly, `App` has `HandleClose` method, but it doesn't seem to be called during shutdown. 
Anyway, added `context.Context` as a parameter and shutdown codepath